ronan_solutions/most_common_element_impl.py

```diff
@@ -8,6 +9,28 @@ def most_common_element(elements: List[Any]) -> List[Any]:
     there are three 0s, three 3s, two 2s, one 1 and one 4, so the correct solution is [0, 3]
     or [3, 0].
     """
-    return [None]
+    #I created a dictionary because I wanted to store not only the number of times a value appears, 
+    # but also the value itself at the same time
+    countPerItem = {}
+    #iteration over the list
+    for i in List:
+            #count the number of times a number appears in the list and attach that as the key to the value that reflects the 
+            #original item
+            countPerItem.append(i,list.count(i))
```

Comment:
There is no `append` function on a dict. To add an element, just do:
```python
countPerItem[key] = value
```

Also: you don't want to use the `count` method on the input list, because this is going to go back over the whole list. If you traverse the whole list for every element you consider in the list, that's going to be N*N steps -- too long! 

Instead, what's useful about the dict is that we can keep track of each time we see an element, so how about doing the following:

```python
    if i not in counterPerItem:
        # If we haven't seen this element yet, put it in the dict with a value of 1
        countPerItem[i] = 1
    else:
        # if we have seen the element, it will have an integer in the dict already,
        # which we can just increment
        countPerItem[i] += 1
```

This we, the dictionary keeps track of the count and updates as we go through the list